### PR TITLE
Remove txn snark local state from protocol state

### PR DESCRIPTION
**Pull request by mrmr1993**
This PR removes `Local_state.t` from the registers exposed by the blockchain snark.

The local state is only ever equal to `Local_state.empty ()` in valid blocks, so exposing it from the blockchain snark is adding to block size and increasing the size of the blockchain snark unnecessarily.

Checklist:

- [ ] Modified the current draft of release notes with details on what is completed or incomplete within this project
- [ ] Document code purpose, how to use it
  - Mention expected invariants, implicit constraints
- [ ] Tests were added for the new behavior
  - Document test purpose, significance of failures
  - Test names should reflect their purpose
- [ ] All tests pass (CI will check this if you didn't)
- [ ] Serialized types are in stable-versioned modules
- [ ] Does this close issues? List them

### CODE REVIEW

**Review by nholland94 — APPROVED** [review]

File: src/lib/pickles/plonk_checks/scalars.ml [hunk 1]
```diff
@@ -527,16 +527,19 @@ module Tick : S = struct
              let x_7 =
                cell (var (Witness 8, Curr)) + cell (var (Witness 5, Curr))
              in
-             square (cell (var (Witness 11, Curr)))
+             (cell (var (Witness 11, Curr)) * cell (var (Witness 11, Curr)))
```
[the remark this answers is not in the document]
Comment (mrmr1993) [reply src/lib/pickles/plonk_checks/scalars.ml]: It's generated code from rust. Reverted in the following commit.

File: src/lib/mina_state/registers.ml [hunk 2]
```diff
@@ -72,3 +72,91 @@ module Checked = struct
         Local_state.Checked.equal' (Field.get f t1) (Field.get f t2) @ acc )
     |> Impl.Boolean.all
 end
+
+module Blockchain = struct
+  type ('ledger, 'pending_coinbase_stack, 'local_state) full_poly =
```
[the remark this answers is not in the document]
Comment (mrmr1993) [reply src/lib/mina_state/registers.ml:77]: Sure, I would love to :)

**Comment by mrmr1993** [comment]
!ci-build-me

**Comment by mrmr1993** [comment]
!ci-build-me

**Comment by mrmr1993** [comment]
We now need this for 2-pass ledger application. Closing.

**Pull request closed without merging**